phylogenetic/scripts/curate_private_data.py

```diff
@@ -0,0 +1,170 @@
+from typing import Any
+from Bio import SeqIO
+import argparse
+from openpyxl import load_workbook
+from datetime import datetime
+from os import path
+from sys import exit
+
+Sequences = dict[str, SeqIO.SeqRecord]
+Metadata = dict[str, dict[str, str]]
+MetadataHeader = list[str]
+
+# The following seem reasonable to hardcode, as they're central to the current mpox workflows
+DATE_COLUMN = 'date'
+ACCESSION = 'accession'
+REQUIRED_COLUMNS = [ACCESSION, DATE_COLUMN] # TODO XXX, 'strain']
+
+# The following could be better provided via parsing the (S3) metadata here, but I want to keep this script isolated
+# from the Snakemake workflow as much as possible
+RECOMMENDED_COLUMNS = ['country', 'division', 'location']
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Parse metadata (xlsx format) and sequences (FASTA) for integration into our canonical mpox pipelines",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument('--sequences', type=str, required=True, nargs='+', metavar='FASTA', help="input sequences")
+    parser.add_argument('--fasta-header-idx', type=int,
+        help='If FASTA headers are "|" separated, this index (1-based) is the accession')
+    parser.add_argument('--xlsx', type=str, required=True, help="Metadata file (Excel .xlsx format)")
+    parser.add_argument('--remap-columns', type=str, nargs='+', default=[], metavar='old:new', required=False,
+        help="Change column names. Note all column names are converted to lower case.")
+    parser.add_argument('--output-sequences', type=str, default=None, help="output sequences (default is to write to 'data/...' to then be used by our Snakemake workflows)")
+    parser.add_argument('--output-metadata', type=str, default=None, help="output metadata (default is to write to 'data/...' to then be used by our Snakemake workflows)")
```

Comment:
If you were going to keep these output arguments, you might set the default values to the relative paths to plan to use below to let the user know what the actual filenames will be before they run the script.

Reply:
Thanks! I've dropped them here for simplicity. If we expand the remit it's easy to bring them back.